File: snippets/javascript/server-event.js
// CFX Server Event Pattern (JavaScript)
// Listen for a network event from a client. Always validate source.

onNet('myResource:serverEvent', (data) => {
    const src = global.source;

    // Validate the source is a real player
    if (!src || src <= 0) return;

    // Validate data
    if (typeof data !== 'object') return;

    // Your server logic here
    console.log(`Player ${src} triggered event`);
});

File: snippets/lua/server-event.lua
-- CFX Server Event Pattern
-- Register a network event handler on the server side.
-- Always capture source immediately and validate input.

RegisterNetEvent('myResource:serverEvent', function(data)
    local source = source -- capture before any yield

    -- Validate the source is a real player
    if not source or source <= 0 then return end

    -- Validate data types
    if type(data) ~= 'table' then return end

    -- Your server logic here
    print(('Player %d triggered event with data'):format(source))
end)
